tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -332,12 +365,17 @@ def agent_test_suite(self, node: Node, variables: Dict[str, Any], log_path: str,
                         self._collect_node_logs()
 
             except:   # pylint: disable=bare-except
-                # Note that we report the error to the LISA log and then re-raise it. We log it here
-                # so that the message is decorated with the thread name in the LISA log; we re-raise
-                # to let LISA know the test errored out (LISA will report that error one more time
-                # in its log)
-                self._log.exception("UNHANDLED EXCEPTION")
-                raise
+                # Report the error and raise and exception to let LISA know that the test errored out.
+                self._log.exception("TEST FAILURE DUE TO AN UNEXPECTED ERROR.")
+                self._report_test_result(
+                    self.context.image_name,
+                    "Setup",
+                    TestStatus.FAILED,
+                    start_time,
+                    message="TEST FAILURE DUE TO AN UNEXPECTED ERROR.",
+                    add_exception_stack_trace=True)
+
+                raise Exception("STOPPING TEST EXECUTION DUE TO AN UNEXPECTED ERROR.")
```

Comment:
Replaced the re-raise with this exception to eliminate the duplicated error message in the LISA log, and to make clearer that we are stopping the LISA run